homeassistant/components/risco/entity.py

```diff
@@ -29,3 +45,82 @@ async def async_added_to_hass(self):
     def _risco(self):
         """Return the Risco API object."""
         return self.coordinator.risco
+
+
+class RiscoCloudZoneEntity(RiscoCloudEntity):
+    """Risco cloud zone entity base class."""
+
+    _attr_has_entity_name = True
+
+    def __init__(
+        self,
+        *,
+        coordinator: RiscoDataUpdateCoordinator,
+        name: str | None,
+        suffix: str,
+        zone_id: int,
+        zone: Zone,
+        **kwargs: Any,
+    ) -> None:
+        """Init the zone."""
+        super().__init__(coordinator=coordinator, **kwargs)
+        self._zone_id = zone_id
+        self._zone = zone
+        self._attr_name = name
+        device_unique_id = zone_unique_id(self._risco, zone_id)
+        self._attr_unique_id = device_unique_id + suffix
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, device_unique_id)},
+            manufacturer="Risco",
+            name=self._zone.name,
+        )
+
+    @property
+    def extra_state_attributes(self) -> Mapping[str, Any] | None:
```

Comment:
If the zone_id won't change during the entity lifetime, we can set `self._attr_extra_state_attributes` in the init method.